### PR TITLE
Add Support For Copying Capability Payloads

### PR DESCRIPTION
**Pull request by nanopay-arthur**
Note: This allows us to submit only the missing payload data.

This needs to be cherry-picked in Release-v3.17

### CODE REVIEW

File: src/foam/nanos/dig/drivers/DigFormatDriver.js [hunk 1]
```diff
@@ -274,7 +274,7 @@ foam.CLASS({
         Object id = obj.getProperty("id");
         if ( id != null &&
              ! SafetyUtil.isEmpty(id.toString()) ) {
-          FObject old = dao.inX(getX()).find(id);
+          FObject old = dao.inX(x).find(id);
```
Comment (lchanmann) [inline src/foam/nanos/dig/drivers/DigFormatDriver.js]: By reading the comment on line 273, seems like using getX() was intentional.

File: src/foam/nanos/crunch/connection/CapabilityPayload.js [hunk 2]
```diff
@@ -47,5 +48,44 @@ foam.CLASS({
         Validation errors for capability data objects.
       `
     }
-  ]
+  ],
+  methods: [
+      {
+        name: 'copyFrom',
+        type: 'CapabilityPayload',
+        args: [ { name: 'obj', type: 'FObject' } ],
```
Comment (lchanmann) [inline src/foam/nanos/crunch/connection/CapabilityPayload.js:56]: wonder is it the intention here to overload FObject.copyFrom?

Comment (nanopay-arthur) [reply src/foam/nanos/crunch/connection/CapabilityPayload.js:56]: Yes, if we use the regular FObject copyFrom on a CapabilityPayload, the existing capabilityDataObjects map gets completely overwritten and we effectively lose the existing data we have.

File: src/foam/nanos/dig/drivers/DigFormatDriver.js (hunk 1 above)
Comment (nanopay-arthur) [reply src/foam/nanos/dig/drivers/DigFormatDriver.js]: hmm, think we'll need to add an auth check to explicitly check for the read permission here.

**Review by adamfox-nanopay — COMMENTED** [review]
LGTM

**Pull request merged**